### ISSUE TITLE
Invalid Access Token

### ISSUE THREAD

**Issue by Kalumo** [mapbox/mapbox-unity-sdk#1396]
Hello,

I am trying to test the Mapbox SDK in Unity and get the invalid access token error, when trying to set up mapbox. I am assuming that it has something to do with the company network firewall. I am able to access data through the browser such as https://api.mapbox.com/v4/mapbox.streets/page.html?access_token=<token>/40.91/27.95.

Can you provide information as to how to fix this issue?

With kind regards,
Kai

Unity version: 2019.3.0a2 Personal
Mapbox SDK version: 2.0.0

**Comment by chinmaygawande123** [comment]
@Kalumo Actually, mapbox provides an access token when your mapbox account is created. For accessing it login into your mapbox account and go in Accounts tab. There you fill find your default token copy it. Open unity > click on mapbox window> click on Setup or Configure > Paste it and submit. You will be able to use The SDK.

**Comment by Kalumo** [comment]
@chinmaygawande123 Yes, that is essentially what I have been doing. When submitting the key an "invalid" message is returned, which I believe to be related to a firewall issue as stated above. My question is if this issue can be resolved, because the firm firewall cannot be turned off.

**Comment by brnkhy** [comment]
If it's a firewall issue, I'm not sure how we can help with that. Strange that firewall is blocking one api endpoint and not another under same domain. 🤔 
@atripathi-mb @jordy-isaac any ideas?

**Comment by supergra** [comment]
I am likewise unable to get any token validated on one machine while I can on another nearly identical one.

I have two Mac systems with identical setups, on the same network:

MacOS Mojave
Unity versions: 2019.1.2  AND 2019.1.7, both exhibit problem
Mapbox SDK version: 2.0.0
Connected to same wireless network at home, firewall disabled.

1) I've tried multiple tokens from multiple Mapbox. All keys work on the first system, while no keys work on the second system.
2) Tried completely wiping the Mapbox folder and reinstalling
3) Tried rebooting Unity and rebooting computer.

**Comment by atripathi-mb** [comment]
With firewalls, you might have to whitelist the domain for api requests to work properly. Unity api requests are sometimes different than browser requests and could work differently.

**Comment by brnkhy** [comment]
@Kalumo @supergra 
I think I found the issue here. I suspect it's happening if you're an old user and used very old versions of the SDK. 
There's a line in the `FileSource.cs` reading the apikey from system environment (if available) but I coulnd't find any line actually setting it. So I think, we used to cache it there back in the days (for some reason I dont know at the moment) but now it's causing problem at token validation.

Anyway, if you change `FileSource.cs` like this, it should fix the issue;
![TortoiseGitMerge_2020-07-17_16-19-05](https://user-images.githubusercontent.com/1465390/87790632-8b181300-c849-11ea-825c-dc18d6f59815.png)

I'll get a PR ready soon but I want to see if I can find and understand why that line was there in the first place.

**Comment by brnkhy** [comment]
also here's the branch; https://github.com/mapbox/mapbox-unity-sdk/tree/TokenValFix

**Comment by supergra** [comment]
@brnkhy We finally discovered our problem was due to some invisible character (line break?) being included when copy-pasting the API key from the Mapbox website into the token field in the Unity GUI (but only on one user's computer, maybe due to some language setting? User was French). I'm sorry I didn't track the specific issue. It may have been an easily-fixable key validation change. We basically couldn't _visually_ see that the key was wrong, but when we copy-pasted somewhere else we saw the extraneous character. If I figure it out, I'll let you know.
